### PR TITLE
Update p4runtime submodule to opensource and p4rt-ctl changes

### PR DESCRIPTION
**Pull request by n-sandeep**
The reason behind these change is:
Custom p4info.proto/P4runtime.proto is not recommended by the community. Changes are made to re-use MatchAction instead of MatchValueLookup externs which are introduced in a forked version of p4runtime repo.

- This PR includes, pointing p4runtime submodule to opensource p4lang
- Support action programming with specific action_params in p4rt-ctl CLI. This will avoid user to program all available params for an action. 
    Existing command: action=my_control.mir_prof_action(0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,1,1, 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0, 0,0,0,0,0,0,0,0,0)

    Command after the change: action=my_control.mir_prof_action(port_id=1,mode=0,port_dest_type=0, dest_id=1,func_valid=1,store_vsi=1)

### CODE REVIEW

**Comment by ffoulkes** [comment]
1. Are there any other dependencies -- for example, does this code only work for newer versions of the P4SDE? If so, what version? This information needs to be captured and relayed to users.
2. If this truly removes the need for the P4Runtime fork, we should also remove `p4runtime-1.3.0-py3-none-any.whl` from the setup directory and update any documentation that refers to it.

**Review by nupuruttarwar — COMMENTED** [review]
LGTM logically and agree with Derek's comments.

Also, has the PR to change documentation for mirroring raised in p4sde repo already?

**Comment by n-sandeep** [comment]
> 1. Are there any other dependencies -- for example, does this code only work for newer versions of the P4SDE? If so, what version? This information needs to be captured and relayed to users.
> 2. If this truly removes the need for the P4Runtime fork, we should also remove `p4runtime-1.3.0-py3-none-any.whl` from the setup directory and update any documentation that refers to it.

The dependency is something like this:
-> User need to build p4 programs artifacts with a specific compiler option for now, then use the latest SDE which has commit related to these change.
-> If user builds p4 programs artifacts 'without' a specific compiler option, they can still use the older version of the LUT extern table programming.

The older version still works fine, we want to keep it for a atleast one release and then remove older code completely.
Yes, will remove `p4runtime-1.3.0-py3-none-any.whl`  file when we completely remove older code.

Derek, Please let me know what is your take on, having older code for one more release

**Comment by ffoulkes** [comment]
> Derek, Please let me know what is your take on, having older code for one more release

I have no objection, though I'm puzzled by the notion of having older code. This PR switches the p4runtime submodule back to p4lang. The disparity would be between any code we have that depends the customized version of P4Runtime and our code.

**Review by ffoulkes — APPROVED** [review]
LGTM

**Comment by ffoulkes** [comment]
The `p4runtime` submodule pointer in `ipdk-io/networking-recipe` currently points to the upstream repo.

This is a push against `p4lang/p4runtime`, not against `ipdk-io/p4runtime-dev` -- i.e., you're upstreaming these changes to the master p4runtime, not our fork.

Is this what you intended?

**Review by saynb — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "stratum/p4runtime"]
 	path = stratum/p4runtime
-	url = https://github.com/ipdk-io/p4runtime-dev.git
+	url = https://github.com/p4lang/p4runtime.git
```
[the remark this answers is not in the document]
Comment (n-sandeep) [reply .gitmodules]: taken care, now we are pointing to ipdk.io/p4runtime submodule which is upgraded to latest p4lang/p4runtime

**Comment by n-sandeep** [comment]
> > Derek, Please let me know what is your take on, having older code for one more release
> 
> I have no objection, though I'm puzzled by the notion of having older code. This PR switches the p4runtime submodule back to p4lang. The disparity would be between any code we have that depends the customized version of P4Runtime and our code.

with the latest change, we are removing older code as well.

**Review by vsureshkumarp — APPROVED** [review]
Changes looks good to me.

**Pull request merged**